tests/ui/traits/dyn-drop-principal.rs

```diff
@@ -0,0 +1,68 @@
+//@ run-pass
+//@ check-run-results
+
+use std::{alloc::Layout, any::Any};
+
+const fn yeet_principal(x: Box<dyn Any + Send>) -> Box<dyn Send> {
+    x
+}
+
+trait Bar: Send + Sync {}
+
+impl<T: Send + Sync> Bar for T {}
+
+const fn yeet_principal_2(x: Box<dyn Bar>) -> Box<dyn Send> {
```

Comment:
yay thanks for testing this one specifically :3

Reply:
This was actually @Jules-Bertholet :D
I only added a `dyn*` test as asked by lcnr.

Reply:
ty jules :3